src/lib/core/trigger.cc

```diff
@@ -35,6 +35,8 @@
 
 #include <small/region.h>
 
+#include <small/mempool.h>
+
 static void
 trigger_fiber_run_timeout(ev_loop *loop, ev_timer *watcher, int revents)
```

Comment:
What about `trigger_fiber_run`? Isn't it vulnerable to the same issues as normal triggers.

I don't like this function. Since it's used only in one place (`on_shutdown_f`), I think we could inline it there, making sure it's safe against trigger deletion.

Reply:
Yes, it's affected as well.

Fixed. Thanks for noticing!